### DISCUSSION TITLE
Allow partial extension metadata

### DISCUSSION THREAD

**Discussion by adriabonetmrf** [crawler-commons/crawler-commons#456] [Q&A]
Hi team,

I came across this library as I needed to parse sitemaps and it's working great so far, there's only one issue that I have and that's that parsed Extension metadata attributes that evaluate as invalid are discarded regardless if strict or allowPartials flags are enabled (here for example: [NewsHandler](https://github.com/crawler-commons/crawler-commons/blob/master/src/main/java/crawlercommons/sitemaps/sax/extension/NewsHandler.java#L68)).

Seeing that partial content is allowed wouldn't it be a good feature to allow partial extensions too via some kind of flagging? I'm validating well formatted sitemaps that might me missing some metadata elements and there's no way to inject custom handlers either to easily bypass this situation.

What do you think?
Thanks

**Comment by sebastian-nagel** [comment]
Hi @adriabonetmrf,

thanks for the suggestion! This definitely makes sense. It would be also good to look whether other extensions also verify the attributes strictly.

One point: if it's possible could you provide an example (or few ones)? This makes it easier to draft a unit test to verify the solution.

Thanks again,
Sebastian

**Reply by sebastian-nagel** [reply]
Opened #458 to track this request.
